src/Components/Search/SearchBar.tsx

```diff
@@ -175,8 +178,33 @@ export class SearchBar extends Component<Props, State> {
     )
   }
 
+  reportPerformanceMeasurement = () => {
+    const duration = performance.now() - this.performanceStart
+    const deviceType = sd.IS_MOBILE ? "mobile" : "desktop"
+
+    const payload = {
+      type: "timing",
+      name: "autocomplete-search-response",
+      timing: duration,
+      tags: [`device-type:${deviceType}`, "design:rich"],
+    }
+
+    request.post(sd.VOLLEY_ENDPOINT).send({
+      serviceName: "force",
+      payload,
+    })
+  }
+
+  shouldMeasurePerformance = () => {
+    return typeof performance !== "undefined" && sd.VOLLEY_ENDPOINT
+  }
+
   searchTextChanged = (_e, { newValue: term }) => {
     this.setState({ term })
+
+    if (this.shouldMeasurePerformance()) {
```

Comment:
Why not put these lines (and the reporting) right in the Relay refetch call? Here it looks like they are lumped into some render/callbacks/side-effect-y places.

I think it can just go in https://github.com/artsy/reaction/blob/6e3144e252f646cf1d45ea2b492b87752948c0b8/src/Components/Search/SearchBar.tsx#L148 and then you probably don't need the `private performanceStart` declaration either (can just be a local in there).